bin/alpen-bridge/src/mode/operator.rs

```diff
@@ -261,3 +325,159 @@ async fn start_rpc_server(
+    s2_client: SecretServiceClient,
+    operator_wallet: &mut OperatorWallet,
+    my_index: OperatorIdx,
+    bitcoin_rpc_client: Arc<BitcoinClient>,
+) {
+    if db
+        .get_pre_stake(my_index)
+        .await
+        .expect("should be able to consult the database")
+        .is_none()
+    {
+        // This means that we don't have a pre-stake tx in the database.
+        // We need to create a pre-stake tx, sign it, broadcast it and save it to the database.
+        info!("no pre-stake tx in the database, creating one");
+        let fee_rate = bitcoin_rpc_client
+            .estimate_smart_fee(1)
+            .await
+            .expect("should be able to get the fee rate estimate");
+
+        let fee_rate = FeeRate::from_sat_per_vb_unchecked(fee_rate);
+
+        info!(?fee_rate, "fee rate");
+
+        // We need to sync the wallet.
+        info!("syncing the operator wallet");
+        operator_wallet
+            .sync()
+            .await
+            .expect("should be able to sync the wallet");
+        info!("synced the operator wallet");
+
+        // Create the PreStake tx.
+        let pre_stake_psbt = operator_wallet
+            .create_prestake_tx(fee_rate)
+            .expect("should be able to create the pre-stake tx");
+        // Get the unsigned pre-stake tx.
+        let pre_stake_tx = pre_stake_psbt.unsigned_tx;
+        let pre_stake_txid = pre_stake_tx.compute_txid();
+        info!(%pre_stake_txid, "created the pre-stake tx");
+
+        // Collect all the UTXOs in the stakechain wallet that match the pre-stake tx inputs.
+        let general_wallet = operator_wallet.general_wallet();
+        let prevouts = pre_stake_tx
+            .input
+            .iter()
+            .map(|i| {
+                let outpoint = i.previous_output;
+                info!(?outpoint, "outpoint");
+                general_wallet
+                    .get_utxo(outpoint)
+                    .expect("should be able to get the outpoint")
+                    .txout
+            })
+            .collect::<Vec<TxOut>>();
+        info!(?prevouts, "prevouts");
+        let num_inputs = prevouts.len();
+        let prevouts = Prevouts::All(&prevouts);
+        let mut sighasher = SighashCache::new(pre_stake_tx);
+        // Sign all the inputs.
+        for input in 0..num_inputs {
+            let sighash = sighasher
+                .taproot_key_spend_signature_hash(input, &prevouts, TapSighashType::Default)
+                .expect("must be able to compute the sighash");
+
+            // Sign the pre-stake tx.
+            let signature = s2_client
+                .general_wallet_signer()
+                .sign(&sighash.to_byte_array(), None)
+                .await
+                .expect("should be able to sign the pre-stake tx");
+
+            sighasher
+                .witness_mut(input)
+                .expect("must be able to get the witness")
+                .push(signature.serialize());
+        }
```

Comment:
@Zk2u I think that this logic is wrong. But I am too tired to reason.
Can you check? I'll try a fresh look tomorrow as well.